classes/class-wc-connect-shipping-label.php

```diff
@@ -226,10 +225,48 @@ public function get_all_items( WC_Order $order ) {
 				}
 
 				$refunded_qty = $order->get_qty_refunded_for_item( $item->get_id() );
+				$remaining_quantity = $item['qty'] - absint( $refunded_qty );
+				/**
+				 * The threshold at which we will start batching items together.
+				 * As an example, if a single order item has a quantity 60, which is more than the default threshold
+				 * value of 20 we will start batching together.
+				 */
+				$threshold = apply_filters( 'wc_connect_shipment_item_quantity_threshold', 20 );
+				/**
+				 * Max number of shipments allowed to be created for this item should quantity of this order items
+				 * exceeds `wc_connect_shipment_item_quantity_threshold`
+				 */
+				$max_shipments = apply_filters( 'wc_connect_max_shipments_if_quantity_exceeds_threshold', 5 );
+				/**
+				 * Default quantity per item if the `wc_connect_shipments_item_threshold_quantity` is not exceeded.
+				 */
+				$quantity_per_shipment = 1;
+
+				$weight_per_item = $item_data['weight'];
+				$should_cap_shipments = $remaining_quantity > $threshold;
+
+				if ( $should_cap_shipments ) {
+					$quantity_per_shipment = floor( $remaining_quantity / $max_shipments );
+					for ( $i = 0; $i < $max_shipments; $i ++ ) {
+						$item_data['weight'] = $quantity_per_shipment * $weight_per_item;
+						$remaining_quantity -= $quantity_per_shipment;
+
+						if( $remaining_quantity > $quantity_per_shipment ) {
```

Comment:
```suggestion
						if( $remaining_quantity >= $quantity_per_shipment ) {
```

Reply:
This fixes a bug when the remaining quantity can be evenly divided by the threshold (e.g. 160):

![screenshot-localhost_8050-2024 05 07-07_33_54](https://github.com/Automattic/woocommerce-services/assets/11618203/64564443-7d52-41cd-bd76-1b113be22147)

Reply:
Once the `$remaining_quantity` gets below the `$quantity_per_shipment`, the weight isn't being calculated correctly. We need to move the `$item_data['weight']` assignment below the `$item_data['quantity']` assignment and use the newly assigned quantity like so:

```
$item_data['weight'] = round($item_data['quantity'] * $weight_per_item, 2);
```